File: CHTL/CHTLNode/CommentNode.cpp
#include "CommentNode.h"
#include "Visitor.h"

namespace CHTL {

CommentNode::CommentNode(const std::string& text) : text(text) {}

void CommentNode::accept(Visitor& visitor) {
    visitor.visitCommentNode(*this);
}

std::unique_ptr<BaseNode> CommentNode::clone() const {
    return std::make_unique<CommentNode>(this->text);
}

} // namespace CHTL

File: src/CHTL/CHTLNode/CommentNode.java
package CHTL.CHTLNode;

/**
 * Represents a comment node in the AST.
 * This is for comments that are intended to be rendered in the output file,
 * such as the '--' generator comments.
 */
public class CommentNode implements BaseNode {
    private final String comment;

    public CommentNode(String comment) {
        this.comment = comment;
    }

    public String getComment() {
        return comment;
    }

    @Override
    public <R> R accept(Visitor<R> visitor) {
        return visitor.visitCommentNode(this);
    }

    @Override
    public BaseNode clone() {
        return new CommentNode(this.comment);
    }
}
